src/Symfony/Component/HttpKernel/Attribute/Cache.php

```diff
@@ -38,27 +38,41 @@ public function __construct(
         public int|string|null $smaxage = null,
 
         /**
-         * Whether the response is public or not.
+         * If true, the contents will be stored in a public cache and serve to all
+         * the next requests.
          */
         public ?bool $public = null,
 
         /**
-         * Whether or not the response must be revalidated.
+         * If true, the response is not served stale by a cache in any circumstance
+         * without first revalidating with the origin.
          */
         public bool $mustRevalidate = false,
 
         /**
          * Additional "Vary:"-headers.
+         *
+         * @var string[]
          */
         public array $vary = [],
 
         /**
          * An expression to compute the Last-Modified HTTP header.
+         *
+         * The expression is evaluated by the ExpressionLanguage component, it
+         * receives all the request attributes and the resolved controller arguments.
+         *
+         * The result of the expression must be a DateTimeInterface.
          */
         public ?string $lastModified = null,
 
         /**
          * An expression to compute the ETag HTTP header.
+         *
+         * The expression is evaluated by the ExpressionLanguage component, it
+         * receives all the request attributes and the resolved controller arguments.
+         *
+         * The result must be a string that will hashed.
```

Comment:
```suggestion
         * The result must be a string that will be hashed.
```
Will it result in a hash or does it need to be a hash already?

Reply:
The listener hashes the result of the expression.
https://github.com/symfony/symfony/blob/d6c879747d19fabbf220ea58b915c3eadf121259/src/Symfony/Component/HttpKernel/EventListener/CacheAttributeListener.php#L72-L75

Reply:
Thanks 👍